diff --git a/withPayloadPlugin.js b/withPayloadPlugin.js
index 1de44bb8029387d7840037109ccd3d5d9983dba4..f08d3b63c5388979751fc13335f40ca6fa8fbd9e 100644
--- a/withPayloadPlugin.js
+++ b/withPayloadPlugin.js
@@ -149,6 +149,7 @@ const withPayload = async (config, paths) => {
             "drizzle-kit/utils": "drizzle-kit/utils",
             "pg-native": "pg-native",
             mongoose: "mongoose",
+            sharp: "commonjs sharp",
           },
         ],
         plugins: [
