diff --git a/node_modules/@payloadcms/next-payload/withPayloadPlugin.js b/node_modules/@payloadcms/next-payload/withPayloadPlugin.js
index 1de44bb..9e327d2 100644
--- a/node_modules/@payloadcms/next-payload/withPayloadPlugin.js
+++ b/node_modules/@payloadcms/next-payload/withPayloadPlugin.js
@@ -60,12 +60,12 @@ const withPayload = async (config, paths) => {
 
   const outputFileTracingExcludes = {
     "/api/**/*": [
-      "./node_modules/@swc/core-linux-x64-gnu",
-      "./node_modules/@swc/core-linux-x64-musl",
-      "./node_modules/@swc/core-darwin-x64",
-      "./node_modules/@swc/core",
-      "./node_modules/@swc/wasm",
-      "./node_modules/@payloadcms/bundler-webpack",
+      // "./node_modules/@swc/core-linux-x64-gnu",
+      // "./node_modules/@swc/core-linux-x64-musl",
+      // "./node_modules/@swc/core-darwin-x64",
+      // "./node_modules/@swc/core",
+      // "./node_modules/@swc/wasm",
+      // "./node_modules/@payloadcms/bundler-webpack",
       ...(config.experimental &&
       config.experimental.outputFileTracingExcludes &&
       config.experimental.outputFileTracingExcludes["/api/**/*"]
@@ -144,11 +144,12 @@ const withPayload = async (config, paths) => {
         externals: [
           ...(incomingWebpackConfig.externals || []),
           {
-            "@swc/core": "@swc/core",
+            // "@swc/core": "@swc/core",
             "drizzle-kit": "drizzle-kit",
             "drizzle-kit/utils": "drizzle-kit/utils",
             "pg-native": "pg-native",
             mongoose: "mongoose",
+            sharp: 'commonjs sharp',
           },
         ],
         plugins: [
